libs/alloc/src/vec/mod.rs

```diff
@@ -4063,7 +4063,12 @@ impl<T, A: Allocator, const N: usize> TryFrom<Vec<T, A>> for [T; N] {
         // We checked earlier that we have sufficient items.
         // The items will not double-drop as the `set_len`
         // tells the `Vec` not to also drop them.
-        let array = unsafe { ptr::read(vec.as_ptr() as *const [T; N]) };
-        Ok(array)
+        // let array = unsafe { ptr::read(vec.as_ptr() as *const [T; N]) };
+        // Ok(array)
```

Comment:
I'd favor either removing this code (as it's no longer used) or prefacing this with a comment saying why it's commented out.